_posts/2021-05-10-helm-charts-for-wildfly.adoc

```diff
@@ -0,0 +1,226 @@
+
+## MicroProfile Application Example
+
+We will use one WildFly quickstart as an example: https://github.com/wildfly/quickstart/tree/master/microprofile-config[microprofile-config].
+We will not cover the code of this example but the important things to mention are:
+
+* This application will be running as a https://docs.wildfly.org/bootablejar/[WildFly Bootable Jar]
+* It uses MicroProfile API and provides a HTTP endpoint. A WildFly server will be provisioned with the `jaxrs-server` and `microprofile-platform` layers so that server contains all the bits (and only those!) to run these features.
+* Its HTTP endpoint uses MicroProfile Config to read a config property from the `CONFIG_PROP` environment variable and returns its value in the HTTP response.
+
+## Install Helm and have access to an OpenShift cluster.
+
+Please refer to https://helm.sh/docs/intro/install/[Installing Helm] page to install Helm in your environment.
+
+We will use https://developers.redhat.com/developer-sandbox[Developer Sandbox for Red Hat OpenShift] as our OpenShift cluster and make sure that we are logged in the cluster and that the `oc` command applies to that cluster.
+
+## Install Helm Repository for WildFly Chart
+
+The `wildfly` Chart can be installed from the https://docs.wildfly.org/wildfly-charts/ repository
+
+[source,nowrap]
+----
+$ helm repo add wildfly https://docs.wildfly.org/wildfly-charts/
+"wildfly" has been added to your repositories
+$ helm search repo wildfly
+NAME                    CHART VERSION   APP VERSION     DESCRIPTION
+wildfly/wildfly         1.3.0           23.0            Build and Deploy WildFly applications on OpenShift$ helm search repo wildfly
+----
+
+The first thing to notice is that the `APP VERSION` points to `23.0`. By default, the Chart will deploy your application using the latest WildFly 23.0 release (23.0.2.Final at the time of this writing).
+
+## Install the Application on OpenShift
+
+Once the `wildfly` Chart is added, we can instal a Helm *release* by using the `helm install` command with the name of the release as well as a YAML configuration file that contains all the settings to build and deploy the application:
+
+[source]
+----
+$ helm install microprofile-config-app \
+    wildfly/wildfly \
+    -f https://raw.githubusercontent.com/wildfly/wildfly-charts/main/examples/microprofile-config/microprofile-config-app.yaml
+
+NAME: microprofile-config-app
+LAST DEPLOYED: Tue May  4 14:52:02 2021
+NAMESPACE: jmesnil1-dev
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+NOTES:
+Your WildFly appplication is building! To follow the build, run:
+
+$ oc get build -w
+
+Note that your Deployment will report "ErrImagePull" and "ImagePullBackOff" until the build is complete. Once the build is complete, your image will be automatically rolled out.
+
+To follow the deployment of your application, run:
```

Comment:
This is not part of the command output but it is surrounded by the [source]

Reply:
that's confusing but it's part of the command output :) 

When you do a `helm install` the command output will show this text to let the users know the next steps to watch their applications.

Reply:
I've reworded the text below to avoid this confusion, thanks!

Reply:
ahhh, ok, sorry for the mistake :) I interpreted it as part of the blog